### PR TITLE
Set cwd to project root and handle errors

### PR DESCRIPTION
**Pull request by ganemone**
Setting the cwd to the project root allows for doing things like running package.json scripts on file changes. Also adds better error logging for failed commands.

### CODE REVIEW

**Comment by ganemone** [comment]
cc @bmingles - any comments?

File: src/extension.ts [hunk 1]
```diff
@@ -60,6 +60,9 @@ class RunOnSaveExtension {
 			var child = exec(cfg.cmd, this._execOption);
 			child.stdout.on('data', data => this._outputChannel.append(data));
 			child.stderr.on('data', data => this._outputChannel.append(data));
+			child.on('error', (e) => {
+				this.showOutputMessage(e);
```
Comment (bmingles) [inline src/extension.ts]: This shows a TypeScript compiler error. You'll probably want to wrap this in `String()`

File: src/extension.ts [hunk 2]
```diff
@@ -80,9 +83,10 @@ class RunOnSaveExtension {
 	}
 
 	private get _execOption(): {shell: string} {
-		if (this.shell) {
-			return {shell: this.shell};
-		}
+		return {
+			shell: this.shell,
+			cwd: vscode.workspace.rootPath
```
Comment (bmingles) [inline src/extension.ts]: This causes a TypeScript compiler error due to a mismatch in the return type.

Comment (ganemone) [reply src/extension.ts]: Fixed

File: src/extension.ts (hunk 1 above)
Comment (ganemone) [reply src/extension.ts]: Fixed

**Pull request merged**